src/redux/selectors/blockchainSelectors.js

```diff
@@ -12,12 +12,12 @@ const allWalletsSelector = createSelector(
 );
 const userWalletAddressSelector = createSelector(
   blockchainReducer,
-  (reducer) => reducer.userWalletAddress,
+  (reducer) => reducer?.userWalletAddress,
```

Comment:
`reducer` should never be empty. It's set here https://github.com/liberland/liberland_frontend/blob/BLOCKCHAIN-425-fe-enable-usage-without-logging-in/src/redux/reducers/index.js#L27 and it has an initial state: https://github.com/liberland/liberland_frontend/blob/BLOCKCHAIN-425-fe-enable-usage-without-logging-in/src/redux/reducers/blockchain.js#L5

Did you experience it being empty in practice? If yes, this suggest we have some really nasty bug somewhere that overwrites whole blockchain state and we should address it instead of defaulting with `?`. If not, then `?` isn't needed here.

Reply:
yea, nice catch, it should be only for blockChainAddress on user, when it's not logged in.